File: Android/app/src/main/java/com/example/androidforios/app/data/model/Prediction.java
package com.example.androidforios.app.data.model;

import android.os.Parcel;
import android.os.Parcelable;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Date;

/**
 * Created by stephen.barnes on 3/21/14.
 */
public class Prediction implements Parcelable{

    public String stopId;
    public String stopName;
    public Integer stopSeconds;

    public Prediction() {}

    public Prediction(JSONObject jsonData) {
        this.importDataFromJSON(jsonData);
    }

    public void importDataFromJSON(JSONObject jsonData) {
        try {
            this.stopId = jsonData.getString("StopID");
            this.stopName = jsonData.getString("Stop");
            this.stopSeconds = jsonData.getInt("Seconds");
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public static Creator<Prediction> CREATOR = new Creator<Prediction>() {
        @Override
        public Prediction[] newArray(int size) {
            return new Prediction[size];
        }

        @Override
        public Prediction createFromParcel(Parcel source) {
            Prediction prediction = new Prediction();
            prediction.stopId = source.readString();
            prediction.stopName = source.readString();
            prediction.stopSeconds = source.readInt();
            return prediction;
        }
    };

    @Override
    public int describeContents() {
        return super.hashCode();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(stopId);
        dest.writeString(stopName);
        dest.writeInt(stopSeconds);
    }
}

File: iOS-Swift/AndroidForiOS/Data/Prediction.swift
//
//  Prediction.swift
//  AndroidForiOS
//
//  Created by Stephen Barnes on 6/14/15.
//  Copyright © 2015 Stephen Barnes. All rights reserved.
//

import Foundation

public struct Prediction {
    public var stopId:String = ""
    public var stopName:String = ""
    public var stopSeconds:NSInteger = 0
    
    public static func parseData(data: NSDictionary) -> Prediction {
        var importedPrediction:Prediction = Prediction()
        guard let stopId = data.objectForKey("StopID") as? String else {
            return importedPrediction  // TODO: throw
        }
        importedPrediction.stopId = stopId
        
        if let stopName = data.objectForKey("Stop") as? String  {
            importedPrediction.stopName = stopName
        }
        if let stopSeconds = data.objectForKey("Seconds") as? NSNumber {
            importedPrediction.stopSeconds = stopSeconds.integerValue
        }
        
        return importedPrediction        
    }
}
